src/util/tile_request_cache.js

```diff
@@ -80,26 +103,45 @@ export function cachePut(request: Request, response: Response, requestTime: numb
     prepareBody(response, body => {
         const clonedResponse = new window.Response(body, options);
 
-        cacheOpen();
-        if (!sharedCache) return;
-        sharedCache
+        cacheOpen(cacheName);
+        if (!sharedCaches[cacheName]) return;
+        sharedCaches[cacheName]
             .then(cache => cache.put(stripQueryParameters(request.url), clonedResponse))
             .catch(e => warnOnce(e.message));
     });
 }
 
+function getQueryParameters(url: string) {
+    const paramStart = url.indexOf('?');
+    return paramStart > 0 ? url.slice(paramStart + 1).split('&') : [];
```

Comment:
Do you think we can use the [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) here? It seems they are [supported]( https://caniuse.com/mdn-api_urlsearchparams) in all major browsers

Reply:
URLSearchParams doesn't parse a full URL string to get you query parameters so I don't think it would add anything in this case